libpod/runtime_img.go

```diff
@@ -124,7 +126,74 @@ func (r *Runtime) RemoveImage(image *image.Image, force bool) (string, error) {
 		return "", fmt.Errorf("unable to delete %s (must force) - image is referred to in multiple tags", image.ID())
 	}
 
-	return image.ID(), image.Remove(force)
+	err = image.Remove(force)
+	if err != nil && errors.Cause(err) == storage.ErrImageUsedByContainer {
+		if errors.Cause(err) == storage.ErrImageUsedByContainer {
+			storageImage, err2 := r.getStorageImage(image.ID())
+			if err2 == nil {
+				errStorage := rmStorageContainers(force, storageImage, image.GetStore())
+				if errStorage == nil {
+					err = image.Remove(force)
+				} else {
+					fmt.Printf("A container created with Buildah may be associated with this image.\n")
+					fmt.Printf("Remove the container with Buildah or retry using the --force option.\n")
+				}
+			} else {
+				fmt.Printf("Unable to find image %s\n", image.ID())
+			}
+		}
+	}
+	return image.ID(), err
+}
+
+// Remove containers that are in storage rather than Podman.
+func rmStorageContainers(force bool, image *storage.Image, store storage.Store) error {
```

Comment:
I would prefer to see this and subsequent functions take a *image.Image ... then you have the store and storage.Image already.  I say that assuming you at one point already had an *image.Image

Reply:
reworking.  I've passed image.Image to the rmStorageContainers and just the id to one other function rather than the whole image.